src/components/game/graphics/objects/piece/DrawPiece.js

```diff
@@ -0,0 +1,24 @@
+import { drawCell } from "../cell/DrawCell";
+
+export const drawPiece = (canvas, piece, xCellSize, yCellSize) => {
+    // Fill in cells from the coreState current piece.
+    var [x, y] = [0, 0];
+    // TODO: Improve
+    if (piece) {
+        var mainCell = piece.mainCell;
+        if (piece != null) {
+            for (const cell of piece.cells) {
+            [x, y] = [cell[1][0] + piece.cx, cell[1][1] + piece.cy];
```

Comment:
Get a function to destructure iterations through piece.cells, mentioned somewhere above too